src/libraries/Common/src/SourceGenerators/TypeRef.cs

```diff
@@ -1,41 +1,47 @@
 // Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
+using System;
 using System.Diagnostics;
 using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.DotnetRuntime.Extensions;
 
-namespace System.Text.Json.SourceGeneration
+namespace DotnetRuntime.SourceGenerators
 {
     /// <summary>
     /// An equatable value representing type identity.
     /// </summary>
-    [DebuggerDisplay("Name = {Name}")]
+    [DebuggerDisplay("Name={Name}")]
     public sealed class TypeRef : IEquatable<TypeRef>
     {
         public TypeRef(ITypeSymbol type)
         {
-            Name = type.Name;
-            FullyQualifiedName = type.GetFullyQualifiedName();
+            (Name, NamespaceDisplayString, FullyQualifiedDisplayString, MinimalDisplayString) = type.GetIdentifierInfo();
```

Comment:
Rendering all those display strings doesn't come cheap and you only need the FQN to establish type equality. Are these really necessary for the purposes of a "ref" type? You can always store these on an on-demand basis in the spec models and use a `Dictionary<TypeRef, TypeSpec>` to recover them as necessary.